api/versions.go

```diff
@@ -549,6 +601,87 @@ func convertStats(cont *info.ContainerInfo) []v2.ContainerStats {
 	return stats
 }
 
+func convertStatsToV2(cont *info.ContainerInfo) []v2.ContainerStats {
```

Comment:
Sure. I don't like this model either. It is very cumbersome. We should scope out the changes required for refactoring the existing API framework.